kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -230,6 +193,73 @@ def exec_foundry_kompile(
             _LOGGER.info(f'Wrote file: {kcfgs_file}')
 
 
+def _contract_json_paths(foundry_out: Path) -> List[str]:
+    pattern = '*.t.sol/*.json'
+    paths = foundry_out.glob(pattern)
+    json_paths = [str(path) for path in paths]
+    json_paths = [json_path for json_path in json_paths if not json_path.endswith('.metadata.json')]
+    json_paths = sorted(json_paths)  # Must sort to get consistent output order on different platforms
+    return json_paths
+
+
+def _contract_from_json(json_path: str) -> Contract:
+    _LOGGER.info(f'Processing contract file: {json_path}')
+    with open(json_path, 'r') as json_file:
+        contract_json = json.loads(json_file.read())
+    contract_name = json_path.split('/')[-1]
+    contract_name = contract_name[0:-5] if contract_name.endswith('.json') else contract_name
+    return Contract(contract_name, contract_json, foundry=True)
+
+
+def _foundry_to_bin_runtime(
+    definition_dir: Path,
+    profile: bool,
+    contracts: Iterable[Contract],
+    main_module: Optional[str],
+    requires: Iterable[str],
+    imports: Iterable[str],
+) -> KDefinition:
+    kevm = KEVM(definition_dir, profile=profile)
+    empty_config = kevm.definition.empty_config(KSort('KevmCell'))
+
+    modules = []
+    for contract in contracts:
+        module = contract_to_main_module(contract, empty_config, imports=imports)
+        _LOGGER.info(f'Produced contract module: {module.name}')
+        modules.append(module)
+    _main_module = KFlatModule(
+        main_module if main_module else 'MAIN',
+        imports=(KImport(mname) for mname in [_m.name for _m in modules] + list(imports)),
+    )
+    modules.append(_main_module)
+
+    bin_runtime_definition = KDefinition(
+        _main_module.name,
+        modules,
+        requires=(KRequire(req) for req in ['edsl.md'] + list(requires)),
+    )
+
+    return bin_runtime_definition
+
+
+def _foundry_to_claims(
+    definition_dir: Path,
+    profile: bool,
+    contracts: Iterable[Contract],
+    main_module: Optional[str],
+) -> List[Tuple[str, KClaim]]:
+    kevm = KEVM(definition_dir, profile=profile)
+    empty_config = kevm.definition.empty_config(KSort('KevmCell'))
+
+    claims: List[Tuple[str, KClaim]] = []
+    for contract in contracts:
+        claims_module = contract_to_spec_module(contract, empty_config, main_module=main_module)
+        _LOGGER.info(f'Produced claim module: {claims_module.name}')
+        claims.extend((claims_module.name, claim) for claim in claims_module.claims)
```

Comment:
We should instead make a _LOGGER message for each additional claim we add, since we don't actually care about the modules anymore.

Reply:
In fact, I guess that this logic can be moved to `contract_to_spec_module`, which should be renamed to `contract_to_claims` (avoid building a module altogether).

Then it could be renamed to `contract_to_claim_cfgs` or the like, and have it just directly construct KCFGs instead of even returning the KClaim.

Reply:
Done. (Except I didn't move logic between `__main__` and `solc_to_k` for now, but eventually we should reorganize these modules.)